src/python/pants/util/ordered_set.py

```diff
@@ -72,19 +74,28 @@ def __eq__(self, other: Any) -> bool:
             return NotImplemented
         return all(x == y for x, y in itertools.zip_longest(self._items, other._items))
 
-    def union(self, *others: Iterable[T]) -> "_AbstractOrderedSet[T]":
+    def __or__(self: _TAbstractOrderedSet, other: Iterable[T]) -> _TAbstractOrderedSet:  # type: ignore[override]
+        # Differences with AbstractSet: our set union forces "other" to have the same type. That is, while
+        # AbstractSet allows {1, 2, 3} | {(True, False)} resulting in Set[Union[int, Tuple[bool, bool]]], the analogous
+        # for descendants  of _TAbstractOrderedSet is not allowed.
+        # GOTCHA: given higher kinded type _TAbstractOrderedSet[S]:
+        #   if T <: S => _TAbstractOrderedSet[S] => *appears* to perform unification but it doesn't
```

Comment:
Freudian slip - please eliminate scala-isms in the comments - `<:` means nothing here to the average target audience.

Reply:
Yes, sir! Well spotted, thank you. In addition to Scala, I would blame pdf-s too - almost fresh grad here.